File: test_features.py
#!/usr/bin/env python3
"""
VOXA Feature Testing Script
Tests new functionality: avatars, SearchBar, homepage flows
"""

import asyncio
import json
import sys
from datetime import datetime
from playwright.sync_api import sync_playwright, expect

TEST_RESULTS = {
    "timestamp": datetime.now().isoformat(),
    "tests": [],
    "critical_issues": [],
    "warnings": []
}

def log_test(name, passed, details=""):
    """Log test result"""
    result = {
        "name": name,
        "passed": passed,
        "details": details,
        "timestamp": datetime.now().isoformat()
    }
    TEST_RESULTS["tests"].append(result)
    status = "✅ PASS" if passed else "❌ FAIL"
    print(f"{status} | {name}")
    if details:
        print(f"  └─ {details}")

def log_critical(issue):
    """Log critical issue"""
    TEST_RESULTS["critical_issues"].append(issue)
    print(f"🔴 CRITICAL: {issue}")

def log_warning(warning):
    """Log warning"""
    TEST_RESULTS["warnings"].append(warning)
    print(f"🟡 WARNING: {warning}")

def main():
    with sync_playwright() as p:
        print("\n" + "="*70)
        print("🚀 VOXA FEATURE TESTING")
        print("="*70 + "\n")

        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        try:
            # ============================================================
            # TEST 1: Homepage loads
            # ============================================================
            print("\n📄 TEST 1: Homepage Loads")
            print("-" * 70)
            page.goto("http://localhost:3000", wait_until="networkidle", timeout=10000)

            title = page.title()
            log_test("Homepage loads", True, f"Title: {title}")

            # ============================================================
            # TEST 2: Avatars load from Supabase
            # ============================================================
            print("\n🖼️ TEST 2: Avatars Load from Supabase")
            print("-" * 70)

            # Get all img elements
            images = page.locator("img").all()
            log_test("Images exist on page", len(images) > 0, f"Found {len(images)} images")

            # Check for avatar URLs
            avatar_urls = []
            supabase_avatars = []
            dicebear_avatars = []

            for img in images:
                src = img.get_attribute("src")
                alt = img.get_attribute("alt")
                if src:
                    avatar_urls.append(src)
                    if "supabase.co" in src:
                        supabase_avatars.append({"src": src, "alt": alt})
                    elif "dicebear" in src:
                        dicebear_avatars.append({"src": src, "alt": alt})

            log_test(
                "Supabase avatars load",
                len(supabase_avatars) > 0,
                f"Found {len(supabase_avatars)} Supabase avatars"
            )

            log_test(
                "Fallback dicebear avatars exist",
                len(dicebear_avatars) > 0,
                f"Found {len(dicebear_avatars)} dicebear fallbacks"
            )

            # Check avatar visibility
            try:
                creator_card = page.locator("a[href*='/perfil/']").first
                if creator_card:
                    avatar_img = creator_card.locator("img").first
                    is_visible = avatar_img.is_visible()
                    log_test("Creator avatars visible", is_visible)
            except:
                log_warning("Could not verify avatar visibility")

            # ============================================================
            # TEST 3: SearchBar Component
            # ============================================================
            print("\n🔍 TEST 3: SearchBar Component")
            print("-" * 70)

            # Find search input
            search_input = page.locator("input[placeholder*='você tem uma pergunta']")
            search_exists = search_input.count() > 0
            log_test("SearchBar input exists", search_exists)

            if search_exists:
                # Type in search
                search_input.fill("henrique")
                page.wait_for_timeout(500)  # Debounce

                # Check if dropdown appears
                dropdown = page.locator("div:has-text('henrique')").filter(has=search_input)
                log_test("SearchBar focuses and accepts input", True)

                # Clear and verify
                search_input.clear()
                log_test("SearchBar clear works", True)

            # ============================================================
            # TEST 4: Navigation Links
            # ============================================================
            print("\n🔗 TEST 4: Navigation & Links")
            print("-" * 70)

            # Check creator cards
            creator_cards = page.locator("a[href*='/perfil/']").all()
            log_test(
                "Creator cards present",
                len(creator_cards) > 0,
                f"Found {len(creator_cards)} creator cards"
            )

            # Check Login button
            login_link = page.locator("a[href='/login']")
            login_exists = login_link.count() > 0
            log_test("Login link exists", login_exists)

            # Check "Você é criador?" link
            creator_link = page.locator("a[href='/sou-criador']")
            creator_exists = creator_link.count() > 0
            log_test("Creator onboarding link exists", creator_exists)

            # ============================================================
            # TEST 5: Page Performance & Resources
            # ============================================================
            print("\n⚡ TEST 5: Performance & Resources")
            print("-" * 70)

            # Get console messages
            console_messages = []
            def on_console(msg):
                console_messages.append({"type": msg.type, "text": msg.text})

            page.on("console", on_console)

            # Reload to capture console
            page.reload(wait_until="networkidle")

            errors = [m for m in console_messages if m["type"] == "error"]
            warnings = [m for m in console_messages if m["type"] == "warning"]

            log_test("No console errors", len(errors) == 0, f"Found {len(errors)} errors")

            if errors:
                for err in errors[:3]:
                    log_critical(f"Console error: {err['text']}")

            # ============================================================
            # TEST 6: Responsive Design
            # ============================================================
            print("\n📱 TEST 6: Responsive Design")
            print("-" * 70)

            # Test mobile viewport
            page.set_viewport_size({"width": 375, "height": 667})  # iPhone size
            page.goto("http://localhost:3000", wait_until="networkidle")

            # Check if nav is still visible
            nav = page.locator("nav").first
            nav_visible = nav.is_visible()
            log_test("Navigation visible on mobile", nav_visible)

            # Check if creators grid adapts
            creator_grid = page.locator("div[class*='grid']").first
            log_test("Grid layout adapts to mobile", creator_grid.is_visible())

            # Reset viewport
            page.set_viewport_size({"width": 1280, "height": 720})

            # ============================================================
            # TEST 7: External Resources
            # ============================================================
            print("\n🌐 TEST 7: External Resources Loading")
            print("-" * 70)

            page.goto("http://localhost:3000", wait_until="networkidle")

            # Check if Supabase scripts loaded
            supabase_loaded = page.evaluate("""
                () => typeof window !== 'undefined' &&
                      document.body.innerHTML.includes('supabase')
            """)
            log_test("Supabase integration present", supabase_loaded)

            # ============================================================
            # SUMMARY
            # ============================================================
            print("\n" + "="*70)
            print("📊 TEST SUMMARY")
            print("="*70)

            passed = sum(1 for t in TEST_RESULTS["tests"] if t["passed"])
            total = len(TEST_RESULTS["tests"])

            print(f"\n✅ Passed: {passed}/{total}")

            if TEST_RESULTS["critical_issues"]:
                print(f"\n🔴 Critical Issues: {len(TEST_RESULTS['critical_issues'])}")
                for issue in TEST_RESULTS["critical_issues"]:
                    print(f"  • {issue}")

            if TEST_RESULTS["warnings"]:
                print(f"\n🟡 Warnings: {len(TEST_RESULTS['warnings'])}")
                for warning in TEST_RESULTS["warnings"]:
                    print(f"  • {warning}")

            # ============================================================
            # SAVE RESULTS
            # ============================================================
            with open("/tmp/test_results.json", "w") as f:
                json.dump(TEST_RESULTS, f, indent=2)

            print("\n✅ Test results saved to /tmp/test_results.json\n")

        except Exception as e:
            log_critical(f"Test execution failed: {str(e)}")
            print(f"\n❌ Error: {e}\n")

        finally:
            browser.close()

if __name__ == "__main__":
    main()

File: frontend/test_features.js
const { chromium } = require('@playwright/test');
const fs = require('fs');

const RESULTS = {
    timestamp: new Date().toISOString(),
    tests: [],
    critical_issues: [],
    warnings: []
};

function logTest(name, passed, details = "") {
    const result = { name, passed, details, timestamp: new Date().toISOString() };
    RESULTS.tests.push(result);
    const status = passed ? "✅ PASS" : "❌ FAIL";
    console.log(`${status} | ${name}`);
    if (details) console.log(`  └─ ${details}`);
}

function logCritical(issue) {
    RESULTS.critical_issues.push(issue);
    console.log(`🔴 CRITICAL: ${issue}`);
}

function logWarning(warning) {
    RESULTS.warnings.push(warning);
    console.log(`🟡 WARNING: ${warning}`);
}

async function main() {
    console.log("\n" + "=".repeat(70));
    console.log("🚀 VOXA FEATURE TESTING");
    console.log("=".repeat(70) + "\n");

    const browser = await chromium.launch({ headless: true });
    const page = await browser.newPage();

    try {
        // ============================================================
        // TEST 1: Homepage Loads
        // ============================================================
        console.log("\n📄 TEST 1: Homepage Loads");
        console.log("-".repeat(70));

        try {
            await page.goto("http://localhost:3000", { waitUntil: "networkidle", timeout: 10000 });
            const title = await page.title();
            logTest("Homepage loads", true, `Title: ${title}`);
        } catch (e) {
            logTest("Homepage loads", false, e.message);
            logCritical("Server not responding - check if npm run dev is running");
        }

        // ============================================================
        // TEST 2: Avatars Load from Supabase
        // ============================================================
        console.log("\n🖼️ TEST 2: Avatars Load from Supabase");
        console.log("-".repeat(70));

        const images = await page.locator("img").all();
        logTest("Images exist on page", images.length > 0, `Found ${images.length} images`);

        let supabaseAvatars = [];
        let dicebearAvatars = [];

        for (const img of images) {
            const src = await img.getAttribute("src");
            if (src) {
                if (src.includes("supabase.co")) {
                    supabaseAvatars.push(src);
                } else if (src.includes("dicebear")) {
                    dicebearAvatars.push(src);
                }
            }
        }

        logTest("Supabase avatars load", supabaseAvatars.length > 0,
            `Found ${supabaseAvatars.length} Supabase avatars`);

        logTest("Fallback dicebear avatars exist", dicebearAvatars.length > 0,
            `Found ${dicebearAvatars.length} dicebear fallbacks`);

        // Show sample URLs
        if (supabaseAvatars.length > 0) {
            console.log(`  Sample Supabase URL: ${supabaseAvatars[0].substring(0, 80)}...`);
        }

        // ============================================================
        // TEST 3: SearchBar Component
        // ============================================================
        console.log("\n🔍 TEST 3: SearchBar Component");
        console.log("-".repeat(70));

        const searchInputs = await page.locator("input[placeholder*='pergunta']").all();
        const searchExists = searchInputs.length > 0;
        logTest("SearchBar input exists", searchExists);

        if (searchExists) {
            const input = searchInputs[0];
            await input.fill("henrique");
            await page.waitForTimeout(600); // Debounce

            logTest("SearchBar accepts input", true);

            await input.clear();
            logTest("SearchBar clear works", true);
        }

        // ============================================================
        // TEST 4: Navigation & Creator Cards
        // ============================================================
        console.log("\n🔗 TEST 4: Navigation & Links");
        console.log("-".repeat(70));

        const creatorCards = await page.locator("a[href*='/perfil/']").all();
        logTest("Creator cards present", creatorCards.length > 0,
            `Found ${creatorCards.length} creator cards`);

        const loginLink = await page.locator("a[href='/login']").count();
        logTest("Login link exists", loginLink > 0);

        const creatorOnboardLink = await page.locator("a[href*='criador']").count();
        logTest("Creator onboarding link exists", creatorOnboardLink > 0);

        // ============================================================
        // TEST 5: Page Structure & Elements
        // ============================================================
        console.log("\n📐 TEST 5: Page Structure");
        console.log("-".repeat(70));

        const nav = await page.locator("nav").count();
        logTest("Navigation present", nav > 0);

        const footer = await page.locator("footer").count();
        logTest("Footer present", footer > 0);

        const heroSection = await page.locator("h1").count();
        logTest("Hero section with title", heroSection > 0);

        // ============================================================
        // TEST 6: Data Attributes & Accessibility
        // ============================================================
        console.log("\n♿ TEST 6: Accessibility");
        console.log("-".repeat(70));

        const buttonsWithoutText = await page.locator("button:not(:has-text(''))").count();
        const buttonsWithAlt = await page.locator("button[aria-label]").count();

        const imagesWithoutAlt = await page.locator("img:not([alt])").count();
        logTest("Images have alt text", imagesWithoutAlt === 0,
            `Found ${imagesWithoutAlt} images without alt`);

        // ============================================================
        // TEST 7: Console Errors
        // ============================================================
        console.log("\n🐛 TEST 7: Console Errors");
        console.log("-".repeat(70));

        const consoleMessages = [];
        page.on("console", msg => consoleMessages.push({ type: msg.type(), text: msg.text() }));

        await page.reload({ waitUntil: "networkidle" });

        const errors = consoleMessages.filter(m => m.type === "error");
        const warnings = consoleMessages.filter(m => m.type === "warning");

        logTest("No console errors", errors.length === 0, `Found ${errors.length} errors`);

        if (errors.length > 0) {
            errors.slice(0, 3).forEach(err => {
                logCritical(`Console error: ${err.text}`);
            });
        }

        // ============================================================
        // TEST 8: Responsive Design
        // ============================================================
        console.log("\n📱 TEST 8: Responsive Design");
        console.log("-".repeat(70));

        await page.setViewportSize({ width: 375, height: 667 });
        await page.goto("http://localhost:3000", { waitUntil: "networkidle" });

        const navMobile = await page.locator("nav").isVisible();
        logTest("Navigation visible on mobile", navMobile);

        const gridMobile = await page.locator("[class*='grid']").first().isVisible();
        logTest("Grid layout visible on mobile", gridMobile);

        // Reset viewport
        await page.setViewportSize({ width: 1280, height: 720 });

        // ============================================================
        // TEST 9: Links are valid
        // ============================================================
        console.log("\n🔗 TEST 9: Link Validation");
        console.log("-".repeat(70));

        const links = await page.locator("a[href]").all();
        let validLinksCount = 0;

        for (const link of links.slice(0, 10)) {
            const href = await link.getAttribute("href");
            if (href && (href.startsWith("/") || href.startsWith("http"))) {
                validLinksCount++;
            }
        }

        logTest("Links are properly formatted", validLinksCount > 0,
            `Checked ${Math.min(10, links.length)} links`);

        // ============================================================
        // SUMMARY
        // ============================================================
        console.log("\n" + "=".repeat(70));
        console.log("📊 TEST SUMMARY");
        console.log("=".repeat(70));

        const passed = RESULTS.tests.filter(t => t.passed).length;
        const total = RESULTS.tests.length;

        console.log(`\n✅ Passed: ${passed}/${total}`);
        console.log(`❌ Failed: ${total - passed}/${total}`);

        if (RESULTS.critical_issues.length > 0) {
            console.log(`\n🔴 Critical Issues: ${RESULTS.critical_issues.length}`);
            RESULTS.critical_issues.forEach(issue => console.log(`  • ${issue}`));
        }

        if (RESULTS.warnings.length > 0) {
            console.log(`\n🟡 Warnings: ${RESULTS.warnings.length}`);
            RESULTS.warnings.forEach(warning => console.log(`  • ${warning}`));
        }

        // Save results
        fs.writeFileSync("/tmp/test_results.json", JSON.stringify(RESULTS, null, 2));
        console.log("\n✅ Test results saved to /tmp/test_results.json\n");

    } catch (e) {
        logCritical(`Test execution failed: ${e.message}`);
        console.error(`\n❌ Error: ${e}\n`);
    } finally {
        await browser.close();
    }
}

main().catch(e => {
    console.error("Fatal error:", e);
    process.exit(1);
});
